### PR TITLE
#60 chore: redux setup

### PR DESCRIPTION
**Pull request by Vanessa-bitega**
What does this PR do?
This pull request implements the setup of Redux in our client application.

Description of Task to be completed
1. Installed Redux and Redux Toolkit to manage the application state.
2. Created Redux store with the necessary configuration.
3. Defined actions and reducers to handle the application state.
4. Connected Redux store to the application components using the `useSelector` and `useDispatch` hooks.
5. Updated the relevant components to use Redux for state management.

How should this be manually tested?
1. Run the application and perform actions that trigger state changes.
2. Verify that the application behaves correctly and the state is updated as expected.
3. Test different scenarios to ensure the Redux store is working correctly.

Screenshots (if appropriate)
N/A

### CODE REVIEW

**Review by Muhet — CHANGES_REQUESTED** [review]
Can you fix the conflict?

**Review by faid-terence — APPROVED** [review]

**Comment by Vanessa-bitega** [comment]
> Can you fix the conflict?

Done

**Review by ALVINdimpos — APPROVED** [review]

**Review by NTElissa — APPROVED** [review]
good progress

File: src/components/ProductListing.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { useEffect } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { fetchProducts } from "../redux/action/ProductAction";
+
+function ProductListing() {
+  const products = useSelector(
+    (state: { products: { data: any } }) => state.products.data
```
Comment (chanceDira) [inline src/components/ProductListing.tsx:7]: Let's not use any keyword, build an interface for product then use it

**Review by chanceDira — CHANGES_REQUESTED** [review]
Nice work @Vanessa-bitega , just one thing

**Review by chanceDira — APPROVED** [review]

**Review by Muhet — APPROVED** [review]
good job Abewe

**Pull request merged**